src/main/java/org/jenkinsci/plugins/codesonar/CodeSonarPublisher.java

```diff
@@ -98,6 +98,31 @@ public String getVisibilityFilter() {
         return this.visibilityFilter;
     }
 
+    // projectName may be a
+    private String getAid(@Nonnull Run<?,?> run, @Nonnull FilePath workspace, @Nonnull String projectName) throws java.io.IOException {
+        /**
+         * The analysis ID from the aid.txt file
+         */
+        String aid = workspace.asCallableWith(new hudson.FilePath.FileCallable<String>() {
+            @Override
+            public void checkRoles(org.jenkinsci.remoting.RoleChecker roleChecker) throws SecurityException {
```

Comment:
Just as a note. We don't check access to workspace here. Probably should be done.